core_test/carmirror_test.go

```diff
@@ -127,7 +129,7 @@ func NewMockStatusSender(channel chan<- MockStatusMessage, orchestrator Orchestr
 
 func (sn *MockStatusSender) SendStatus(have filter.Filter[mock.BlockId], want []mock.BlockId) error {
 	state := sn.orchestrator.State()
-	sn.channel <- MockStatusMessage{state, have, want}
+	sn.channel <- MockStatusMessage{state, have.Copy(), want}
```

Comment:
Out of curiosity, is this Copy known to be required, or just left over after debugging?

Reply:
Known to be required. There was a condition where the filter was cleared on the sink and that caused the filter to be cleared on the source because they were actually the same object. Obviously that's a slightly artificial case because in the real world the source and sink will not be in the same process and the 'copy' will be done implicitly by serializing and deserializing the message.